File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        long descents = 0, curr = 1;
        for(var i=1; i<prices.length; i++){
            if(prices[i-1]-prices[i] == 1)
                curr++;
            else{
                descents += apSum(curr);
                curr = 1;
            }
        }
        return descents + apSum(curr);
    }
    
    private long apSum(long n){
        return (n*(n+1))/2;
    }
}

File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.py
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        descents, curr = 0, 1
        for i in range(1, len(prices)):
            if prices[i - 1] - prices[i] == 1:
                curr += 1
            else:
                descents += self.ap_sum(curr)
                curr = 1

        return descents + self.ap_sum(curr)

    def ap_sum(self, n):
        return n * (n + 1) // 2   